src/ringct/rctOps.cpp

```
@@ -219,6 +220,32 @@ static const zero_commitment zero_commitments[] = {
 
 namespace rct {
 
+    // ── X generator (asset ID blinding base, HF21+) ──────────────────────────
+    // X = 8 * hash_to_curve("beldex_asset_id_blinding_generator")
+    // Computed once at first use; same construction as ge_p3_H.
+    static ge_p3 s_ge_p3_X;
+    static std::once_flag s_ge_p3_X_flag;
+
+    static void init_ge_p3_X() {
+        static const char domain[] = "beldex_asset_id_blinding_generator";
```

Clang (14.0.6)

[warning] do not declare C-style arrays, use std::array<> instead

(modernize-avoid-c-arrays)